test/clj/athenaeum/handlers/user_test.clj

```diff
@@ -0,0 +1,72 @@
+(ns athenaeum.handlers.user-test
+  (:require [clojure.test :refer :all]
+            [athenaeum.fixtures :as fixtures]
+            [athenaeum.handlers.user :as user]
+            [athenaeum.domain.user :as domain-user]
+            [athenaeum.session :as session]
+            [athenaeum.test-utils :as tu]
+            [athenaeum.db :as db]))
+
+(use-fixtures :once fixtures/load-config fixtures/set-datasource fixtures/set-redis-server-conn)
+(use-fixtures :each fixtures/clear-tables fixtures/clear-sessions)
+
+(deftest login-test
+  (testing "On successful id token verification, creates session,
+            adds id-token cookie to response and returns status 200"
+    (let [req {:headers {:id-token "valid-id-token"}}
+          res (with-redefs [user/verify-id-token (constantly true)
+                            user/get-payload (constantly {:google-id "google-id"
+                                                          :name      "name"
+                                                          :email     "name@nilenso.com"
+                                                          :domain    "nilenso.com"
+                                                          :photo-url "picture"})
+                            session/new-id (constantly "new-session-id")]
+                (user/login req))]
+      (is (= true (session/exists? "new-session-id")))
+      (is (= "new-session-id" (get-in res [:cookies :session-id :value])))
+      (is (= 200 (:status res)))))
+
+  (testing "If email domain is not nilenso.com, returns status 401 "
+    (tu/with-fixtures
+      [fixtures/clear-tables fixtures/clear-sessions]
+      (let [req {:headers {:id-token "valid-id-token"}}
+            res (with-redefs [user/verify-id-token (constantly true)
+                              user/get-payload (constantly {:google-id "google-id"
+                                                            :name      "name"
+                                                            :email     "name@foo.bar"
+                                                            :domain    "foo.bar"
+                                                            :picture   "picture"})
+                              session/new-id (constantly "new-session-id")]
+                  (user/login req))]
+        (is (= 401 (:status res)))
```

Comment:
This should return a `400`, not a `401`. `401` means that the failure was because the client needs to authenticate to access the resource, which doesn't make sense for a login API. `400` means that there was bad or invalid data in the request.